streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -1012,6 +1022,8 @@ private void prepareChangelogs(final Map<TaskId, Task> tasks,
                 // no records to restore; in this case we just initialize the sensor to zero
                 final long recordsToRestore = Math.max(changelogMetadata.restoreEndOffset - startOffset, 0L);
                 task.recordRestoration(time, recordsToRestore, true);
+            }  else if (changelogMetadata.stateManager.taskType() == TaskType.STANDBY && storeMetadata.endOffset() != null) {
+                standbyUpdateListener.onUpdateStart(partition, storeName, startOffset, storeMetadata.endOffset());
```

Comment:
I notice we wrap the restore listener in a try-catch block, let's do the same for the standby listener. Maybe also log an error saying which listener threw the error inside the `catch` block in both cases?